commit 979f5b73cff85779e140d84888729e8f4e2ea74e (refs/changes/62/2662/1)
Author: JoÌˆrn Zaefferer <joern.zaefferer@gmail.com>
Date:   2017-09-27T11:01:51-07:00 (2 years ago)
    
    joyent/node-docker-registry-client#24 joyent/node-docker-registry-client#23 Namespace validation too strict

diff --git a/lib/common.js b/lib/common.js
index ce77d12..89c8be5 100644
--- a/lib/common.js
+++ b/lib/common.js
@@ -33,7 +33,7 @@ var DEFAULT_LOGIN_SERVERNAME = 'https://index.docker.io/v1/';
 // 'DEFAULTTAG' from https://github.com/docker/docker/blob/0c7b51089c8cd7ef3510a9b40edaa139a7ca91aa/graph/tags.go#L25
 var DEFAULT_TAG = 'latest';
 
-var VALID_NS = /^[a-z0-9_-]*$/;
+var VALID_NS = /^[a-z0-9\._-]*$/;
 var VALID_REPO = /^[a-z0-9_\/\.-]*$/;
 
 
@@ -203,7 +203,7 @@ function parseRepo(arg, defaultIndex) {
         }
         if (! VALID_NS.test(ns)) {
             throw new Error('invalid repository namespace, may only contain ' +
-                '[a-z0-9_-] characters: ' + ns);
+                '[a-z0-9._-] characters: ' + ns);
         }
         if (ns[0] === '-' && ns[ns.length - 1] === '-') {
             throw new Error('invalid repository namespace, cannot start or ' +
diff --git a/test/common.test.js b/test/common.test.js
index 2ebb4d2..e420eac 100644
--- a/test/common.test.js
+++ b/test/common.test.js
@@ -363,6 +363,20 @@ test('parseRepo', function (t) {
         'canonicalName': 'myreg.example.com:1234/foo/bar'
     });
 
+    t.throws(function () { parseRepo('registry.gitlab.com/user@name/repo-a/repo-b'); },
+        /invalid repository namespace/);
+
+    t.deepEqual(parseRepo('registry.gitlab.com/user.name/repo-a/repo-b'), {
+        'index': {
+            'name': 'registry.gitlab.com',
+            'official': false
+        },
+        'official': false,
+        'remoteName': 'user.name/repo-a/repo-b',
+        'localName': 'registry.gitlab.com/user.name/repo-a/repo-b',
+        'canonicalName': 'registry.gitlab.com/user.name/repo-a/repo-b'
+    });
+
     t.end();
 });
 
